src/utils/theme.ts

```diff
@@ -69,16 +69,46 @@ export function initTheme() {
     darkModeToggle.addEventListener("click", () => {
         toggleDarkMode();
     });
+
+    window
+        .matchMedia("(prefers-color-scheme: dark)")
+        .addEventListener("change", setDarkTheme);
     setColorScheme();
 }
 
+function setThemeFromPreference() {
+    if (
+        window.matchMedia &&
+        window.matchMedia("(prefers-color-scheme: dark)").matches
+    ) {
+        darkModeOn();
+    } else {
+        darkModeOff();
+    }
+}
+
+function setDarkTheme(event) {
```

Comment:
This seems to get by the linter but `event` needs to be typed properly in order to build.

Reply:
I looked into this, but I couldn't figure out what type this should be. Lack of skill problem.

Reply:
I made it an arrow function again, so I don't need to think about it. I made this a real function because I wanted to remove the event listener. So no need for this anymore.